### ISSUE TITLE
Dispatched not working

### ISSUE THREAD

**Issue by Pfeifenjoy** [elierotenberg/react-prepare#1]
Hey,

I just found this project, and would really like to use it in my private project.
However I have some trouble with the API.

What is the difference between prepared and dispatched.

The documentation says:
Helper to use prepared more simply if your side effects consists mostly of dispatching redux actions.

As far as I understand this just the arguments of the callback should be different.
However only the prepared decorator works for me.

It seems that the nextPrepare function doesn't get the second parameter:
https://github.com/elierotenberg/react-prepare/blob/master/src/dispatched.js#L5

Somehow in the prepared module the function prepare is only called with one argument.
https://github.com/elierotenberg/react-prepare/blob/master/src/prepared.js#L17
https://github.com/elierotenberg/react-prepare/blob/master/src/prepared.js#L23

To give you some feedback.
I think the props in the prepared decorator are enough, because for me they include the 
dispatch function.
Also an option to only call the callback function once would be very nice.
I ran into the problem to dispatch an action which updated my store and reran the callback function
which resulted in a loop.

**Comment by elierotenberg** [comment]
Hello,

Thanks for your helpful feedback.
Would you care for submitting a pull request?
If not, I can try and fix the problem myself. I would also appreciate an example of code not working the way it is supposed to, so I can add a minimal test checking for this :)

**Comment by Pfeifenjoy** [comment]
Hey,

My component is basically something like this:
```
import React, { Component } from "react"
import { connect } from "react-redux"
import { withRouter } from "react-router"
import { load } from "../actions/"
import { dispatched } from "react-prepare"

@connect(({ items }) => ({ items }))
@withRouter
@dispatched(props => props.dispatch(load())) //or @dispatched((props, dispatch) => dispatch(load()))
export default class MyComponent extends Component {
    render() {
        return <div>...</div>
    }
}
```

I already thought about creating a pull request, where I would pass in the dispatch function or the context including the store:
https://github.com/elierotenberg/react-prepare/blob/master/src/dispatched.js#L5

But I was not sure about what was supposed to be passed in the nextPrepare function as a second argument.

Also if I use the connect decorator, I will get the dispatch function in the props.
Then I don't need it as a second argument.
That's why I got confused about the difference between dispatched and prepared.

So I was not sure if I should create a pull request or just use the prepared decorator.
Basically something like this should be enough assuming that the connect decorator is used:

const nextPrepare = (props) => prepare(props, props.dispatch);

**Comment by Pfeifenjoy** [comment]
In my PR I am assuming you wanted to pass the context.
I thought this makes the most sense with the prepared decorator.

**Comment by elierotenberg** [comment]
Would you please tell me if 3c5fe53 fixes this?
